lib/interactor/initializer/initialize.rb

```diff
@@ -3,33 +3,57 @@ def self.for(*args)
     new(*args).run
   end
 
-  attr_reader :target_class, :attributes, :keyword_params
+  attr_reader :target_class, :attributes, :defaults, :keyword_params
 
-  def initialize(target_class, attributes, keyword_params: false)
+  def initialize(target_class, attributes, defaults, keyword_params: false)
     @target_class = target_class
     @attributes = attributes
+    @defaults = defaults
     @keyword_params = keyword_params
   end
 
   def run
+    define_initializer
+    define_defaults_initializer
+  end
+
+  private
+
+  def define_initializer
     target_class.class_eval <<-RUBY
-      def initialize(#{initializer_signature(attributes)})
-        #{attr_assignments(attributes)}
+      def initialize(#{initializer_signature})
+        #{attr_assignments}
+        initialize_defaults
       end
     RUBY
   end
 
-  private
+  def define_defaults_initializer
+    target_class.class_exec(defaults) do |defaults|
+      define_method :initialize_defaults do
+        defaults.keys.each do |name|
+          instance_variable_set("@#{name}", defaults[name]) unless instance_variable_get("@#{name}")
```

Comment:
So if you call the interactor with `.for(value: nil)` it would fall back to the default? Doesn't seem right.